opencog/cogserver/server/CogServer.cc

```diff
@@ -234,6 +264,10 @@ void CogServer::runLoopStep(void)
                   );
     }
 
+
+    // Process asynch requests
+    processAsynchRequests();
```

Comment:
Yeah, please don't do this.  The listen socket is in a distinct thread from the request queue, so this is not needed.   The requests need to be processed in order, not out of order.  If some request wishes to launch a thread, it can do so on it's own.  We should not default this behavior.

Anyway, this entire block of code is deprecated anyway; I want to make requests and the cogserver go away; it doesn't make sense to hack on this code.